sparsnas_decode.cpp

```diff
@@ -191,18 +189,30 @@ class SignalDetector {
 //      Note that data_[4] cycles between 0-3 when you first put in the batterys in t$
         if(data4 == 1){
           watt = (float)((3600000 / PULSES_PER_KWH) * 1024) / (effect);
-        }      
-        m += sprintf(m, "{\"Sequence\":\"%5d\",\"Watt\":\"%7.1f\",\"kWh\":\"%d.%.3d\",\"battery\":\"%d\",\"FreqErr\":\"%.2f\"}", seq, watt, pulse/PULSES_PER_KWH, pulse%PULSES_PER_KWH, battery, freq);
+        }
+        m += sprintf(m, "{\"Sequence\": %5d,\"Watt\": %7.2f,\"kWh\": %d.%.3d,\"battery\": %d,\"FreqErr\": %.2f", seq, watt, pulse/PULSES_PER_KWH, pulse%PULSES_PER_KWH, battery, freq);
         if (testing && crc == packet_crc) {
           error_sum += fabs(freq);
           error_sum_count += 1;
         }
       }
 
-      m += sprintf(m, (crc == packet_crc) ? "\n" : "CRC ERR\n");
+      m += sprintf(m, (crc == packet_crc) ? ",\"CRC\":\"ok\"" : ",\"CRC\": \"ERR\"");
 
+      m += sprintf(m, ",\"Sensor\":%6d}\n", SENSOR_ID);
       if (!testing) {
-        fprintf(stderr, "%s", mesg);
+        if (mosq && !bad) {
+          int ret = mosquitto_publish (mosq, NULL, MQTT_TOPIC, strlen(mesg) - 1, mesg, 0, false);
+          if (ret) {
```

Comment:
@bhaap

I think it would be better to insert this code below line `205`:

```c++
if ( ret != MOSQ_ERR_SUCCESS) {
  mosquitto_reconnect(mosq);
  ret = mosquitto_publish (mosq, NULL, MQTT_TOPIC, strlen(mesg) - 1, mesg, 0, false);
}
```